File: Question2/Question2.py
def solution(files):
    # Write solution here
    counter_letter= {}
    totalOddLet=0
    oddLet=[]
    len_files=len(files)
    for i in range(len_files):
        if files[i] not in counter_letter.keys():
            counter_letter[files[i]] = 1
        else:
            counter_letter[files[i]] += 1
    for x in counter_letter.keys():
        if (counter_letter[x] % 2 != 0):
            oddLet.append(x)
            totalOddLet += 1
    while  len_files!=0:
        if totalOddLet >1 or totalOddLet%2!=1 and  len_files%2 == 0:
            len_files -=1
            totalOddLet -= 1
        else:
            return len_files


def main():
    str = input()
    answer = solution(str)
    print(answer)


if __name__ == '__main__':
    main()

File: Question2/Question2.cpp
#include <bits/stdc++.h>
using namespace std;
int solution(string files) { 
    int len_files = files.length();
    if (len_files==0)
      return -1;
    else if (len_files==1)
      return 1;
    int totalOddLet = 0;
    unordered_map<char, int> counter_letter;
    for (auto c:files) {
        if (counter_letter.find(c) == counter_letter.end()) 
            counter_letter[c] = 1;
         else 
            counter_letter[c] += 1;
    }
    for (auto x : counter_letter) {
        if (x.second % 2 != 0) 
            totalOddLet += 1;
    }
    while (len_files != 0) {
        if ( (totalOddLet % 2 != 1 && len_files % 2 == 0) || totalOddLet > 1) {
            len_files--;
            totalOddLet--;
        } else 
             return len_files;
    }
    return 0;}
int main() {
     string files="";
     cin>>files;
     cout<<solution(files);
     return 0;
}